doc/man/wesnothd.6

```diff
@@ -49,6 +49,7 @@ on what commands the server accepts via the wesnoth client (/query ...) or the f
 tells wesnothd where to find the config file to use. See the section
 .B SERVER CONFIG
 below for the syntax. You can reload the config with sending SIGHUP to the server process.
+When used as a LAN server, the 'command' attribute can be used to where to specify either this executable or a script file to run this executable.
```

Comment:
This confuses me. What does it have to do with the changes made in this PR?

Reply:
That's part of what the PR does?

Reply:
What? There isn't anything in this PR that changes the server though?

Reply:
It adds an attribute that can be present in the server config that gets passed to wesnothd when the wesnoth client starts it as a LAN server. I don't mind putting that documentation somewhere else, but I couldn't think of any better place.

Reply:
Hmm. It seems I misunderstood something. The place you added `command` was not the place I thought you added it. I was thinking of a tag that is probably part of the preferences file, though I don't see it in my own preferences file so it's hard to check…

I now understand what soliton was talking about. It does seem very odd for this config file to contain the `command` key.

Reply:
The preferences already has an attribute for that, so it wouldn't make sense to add a second attribute for it. The other alternative of course still being to add a define specifically for the wesnothd location.

Reply:
I don't understand what you're saying… the preferences already has the thing you added? So why did you add it if it already exists? I'm _really_ confused now.

Reply:
The current functionality allows specifying a "system preferences file" that can be used to provide default values for preferences.

---

The use case for this functionality was apparently to:
1) Locate the wesnothd executable when it's been given a different name than the Wesnoth client expects when packaging for Linux distros. This is set using the `mp_server_program_name` preference attribute.
2) Specify some other server settings such as the port to use - this was apparently never actually implemented.

---

This PR removes that "system preferences file" handling using the `mp_server_program_name` preferences attribute and replaces it with the `command` attribute that Linux packagers can append to [lan_server.cfg](https://github.com/wesnoth/wesnoth/blob/master/data/lan_server.cfg).

---

The result being that for any packager who's using this (and so far only the Arch packager has replied to the email I sent to the packager's ML to say he doesn't use it for Arch), instead of creating a "system preferences file" containing just `mp_server_program_name=/path/to/wesnothd` and also compiling Wesnoth with the `DEFAULT_PREFS_PATH=/path/to/system/preferences` they would instead just append `command=/path/to/wesnothd` to `data/lan_server.cfg`.

Reply:
It is definitely a weird place to put that information. Preferences or GameConfigWML seem like a better option to me. Not sure how viable the latter would actually be for packagers, mind you.

Reply:
It's not that weird - it's putting the location for where to find wesnothd for LAN in the file with other LAN configuration settings. IMO that makes a lot more sense than having to setup a separate preferences file and then build Wesnoth with a compile option pointing to that file.

Reply:
Except for the fact that that file is intended for settings to configure the server once it has started, not settings that the client uses to find the server.

Reply:
I don't think that's a reason it can't be used for this though.